src/table/utils/color-utils.ts

```diff
@@ -43,29 +46,37 @@ export function resolveToRGBAorRGB(input) {
   return 'none';
 }
 
-export function isDarkColor(color) {
+export function isDarkColor(color: string | undefined): boolean {
+  if (color === undefined) return false;
+
   const rgba = resolveToRGBAorRGB(color);
   const matches =
     /^rgb\(\s*(\d{1,3})\s*,\s*(\d{1,3})\s*,\s*(\d{1,3})\s*\)$/i.exec(rgba) ||
     /^rgba\(\s*(\d{1,3})\s*,\s*(\d{1,3})\s*,\s*(\d{1,3})\s*,\s*(\d(\.\d+)?)\s*\)$/i.exec(rgba);
-  const r = +matches?.[1];
-  const g = +matches?.[2];
-  const b = +matches?.[3];
+  const r = matches?.[1];
+  const g = matches?.[2];
+  const b = matches?.[3];
+  const validRGB = r !== undefined && g !== undefined && b !== undefined;
 
   // Using the HSP (Highly Sensitive Poo) value, determine whether the color is light or dark
   // HSP < 125, the color is dark, otherwise, the color is light
-  return 0.299 * r + 0.587 * g + 0.114 * b < 125;
+  return validRGB ? 0.299 * +r + 0.587 * +g + 0.114 * +b < 125 : false;
 }
 
-export function isTransparentColor(color) {
+export function isTransparentColor(color: string | undefined): boolean {
+  if (color === undefined) return false;
+
   const rgba = resolveToRGBAorRGB(color);
   const matches = /^rgba\(\s*(\d{1,3})\s*,\s*(\d{1,3})\s*,\s*(\d{1,3})\s*,\s*(\d(\.\d+)?)\s*\)$/i.exec(rgba);
-  return +matches?.[4] === 0;
+  const a = matches?.[4];
+  return a !== undefined ? +a === 0 : false;
 }
 
-export function removeOpacity(color) {
-  const rgba = resolveToRGBAorRGB(color);
-  const matches = /^rgba\(\s*(\d{1,3})\s*,\s*(\d{1,3})\s*,\s*(\d{1,3})\s*,\s*(\d(\.\d+)?)\s*\)$/i.exec(rgba);
-  if (matches) return `rgb(${matches[1]},${matches[2]},${matches[3]})`;
+export function removeOpacity(color: string | undefined): string | undefined {
+  if (color !== undefined) {
```

Comment:
I decided to take care of the undefined on this level so that `resolveToRGBAorRGB` always receives a string